src/main/java/net/minecraftforge/internal/ForgeBindings.java

```diff
@@ -15,25 +15,31 @@
 
 import java.util.function.Supplier;
 
-public class ForgeBindings implements IBindingsProvider {
+public final class ForgeBindings implements IBindingsProvider {
     @Override
     public Supplier<IEventBus> getForgeBusSupplier() {
         return ()-> MinecraftForge.EVENT_BUS;
     }
 
     @Override
     public Supplier<I18NParser> getMessageParser() {
-        return ()->new I18NParser() {
-            @Override
-            public String parseMessage(final String i18nMessage, final Object... args) {
-                return ForgeI18n.parseMessage(i18nMessage, args);
-            }
-
-            @Override
-            public String stripControlCodes(final String toStrip) {
-                return ForgeI18n.stripControlCodes(toStrip);
-            }
-        };
+        final class LazyInit {
+            private static final Supplier<I18NParser> INSTANCE = () -> new I18NParser() {
```

Comment:
I mean I kinda get the point of making it a lazy init as the inner class ha no captures, But why is this a named class inside this method? Its just ugly code IMO to have a named class inside a method. I'd say move this out to a private inner class instead.